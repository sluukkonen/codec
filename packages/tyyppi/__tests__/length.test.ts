import { length, string } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

const twoNumbers = length(string, 2)

test("successfully parses if the length is exactly the designed length", () => {
  expectParseSuccess(twoNumbers, "ab")
})

test("fails to parse if length property is missing or is not an integer", () => {
  expectParseFailure(twoNumbers, null)
  expectParseFailure(twoNumbers, { length: "0" })
  expectParseFailure(twoNumbers, { length: NaN })
})

test("fails to parse if the length is below the minimum length", () => {
  expectParseFailure(twoNumbers, "a")
})

test("fails to parse if the length is above the maximum length", () => {
  expectParseFailure(twoNumbers, "abc")
})
