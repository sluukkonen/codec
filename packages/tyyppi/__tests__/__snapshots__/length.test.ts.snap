// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fails to parse if length property is missing or is not an integer 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received null",
    "path": [],
    "received": "null",
  },
]
`;

exports[`fails to parse if length property is missing or is not an integer 2`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received an object",
    "path": [],
    "received": "object",
  },
]
`;

exports[`fails to parse if length property is missing or is not an integer 3`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received an object",
    "path": [],
    "received": "object",
  },
]
`;

exports[`fails to parse if the length is above the maximum length 1`] = `
[
  {
    "code": "too_long",
    "length": 3,
    "maxLength": 2,
    "message": "Expected the string to contain at least 2 character(s)",
    "path": [],
    "type": "string",
  },
]
`;

exports[`fails to parse if the length is below the minimum length 1`] = `
[
  {
    "code": "too_short",
    "length": 1,
    "message": "Expected the string to contain at least 2 character(s)",
    "minLength": 2,
    "path": [],
    "type": "string",
  },
]
`;
